src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,233 @@
+        if (taskNumber < taskList.size()) {
+            System.out.println("Okiii... This task has been marked as done");
+            taskList.get(taskNumber).markAsDone();
+            System.out.println((taskList.get(taskNumber)).description);
+        } else {
+            throw new DukeException();
+        }
+        printDashLine();
+    }
+
+    private static void markUnDone (ArrayList <Task> taskList, String input) throws DukeException{
+        int taskNumber = Integer.parseInt(input.substring(input.indexOf(MARK_UNDONE) + MARK_UNDONE.length()));
+        if (taskNumber < taskList.size()) {
+            System.out.println("Okiii... This task has been marked as not done yet");
+            taskList.get(taskNumber).markAsUndone();
+            System.out.println((taskList.get(taskNumber)).description);
+        } else {
+            throw new DukeException();
+        }
+
+        printDashLine();
+    }
+
+    private static void listTasks (ArrayList <Task> taskList){
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.println(Integer.toString(i + 1) + " " + (taskList.get(i)).toString());
+        }
+        printDashLine();
+    }
+
+    public static void main(String[] args) throws IOException {
+        String logo = "  ____                 _     \n" +
+                " |  _ \\ ___  __ _  ___| |__  \n" +
+                " | |_) / _ \\/ _` |/ __| '_ \\ \n" +
+                " |  __/  __/ (_| | (__| | | |\n" +
+                " |_|   \\___|\\__,_|\\___|_| |_|\n" +
+                "                             ";
+        System.out.println(logo);
+
+        String intro = "__________________________________________________ \n"
+                + "  Hello! I'm Peach  \n"
+                + "  What can I do for you? \n"
+                + "__________________________________________________ \n";
+
+        System.out.println(intro);
+
+        if (!(java.nio.file.Files.exists(DATA_STORAGE_PATH))){
+            File file = new File (DATA_STORAGE_PATH.toString());
+            file.getParentFile().mkdirs();
+            System.out.println("Data Storage File successfully created \n");
+        }
+
+        try {
+            Storage.loadTasks();
+        } catch (FileNotFoundException e){
+            System.out.println ("File is not found!");
+        }
+
+        Scanner in = new Scanner(System.in);
+        String input = in.nextLine();
+
+        while (input.equals(BYE) == false) {
+            printDashLine();
+            if (input.equals(LIST)) {
+                listTasks(taskList);
+            } else if (input.contains(MARK_UNDONE)) {
+                try {
+                    markUnDone(taskList, input);
+                }
+                catch (DukeException e)
+                {
+                    System.out.println("The task number is out of bound and therefore cannot be marked undone!");
+                }
```

Comment:
You might want to have a single `try`/`catch` to wrap the whole body of the `while` loop, and have an `enum` for each type of error message (e.g. `UNDONE_EXCEPTION_MSG`, `TASK_EXCEPTION_MSG` etc.) that gets displayed accordingly in the `catch` block.